### DISCUSSION TITLE
Improving decodeLogs

### DISCUSSION THREAD

**Discussion by NBMSacha** [wevm/viem#1637] [Idea / Feature Request]
Hello! 

Been using decodeLogs, switching from decodeEvents of Web3JS due to improve perf (almost x2, congrats!)

I had some ideas in mind which I thought might be worth sharing: 

1. decodeLogs crash sensitivity 
decodeLogs is crashing when trying to parse an event with an ABI which *would work* BUT isn't exactly similar. For example, I'm parsing Withdraw/Deposit events with the same piece of code - the events are identical, only the name differs. With Web3js, I had no crash, but with decodeLogs I have to duplicate / complexify my code while the parsing remains the same.

2. Checksum is expensive
When parsing addresses, Viem is applying Checksum by default - checksum is super expensive, I believe most of the resources of my indexing script (indexing > 1B events) is actually coming (CPU wise) from cheksum events of this decodeLogs, while I don't need this at all, I'm lowercasing all the addresses for harmonization purpose - might be interesting to make it optional, maybe even at the library level - i.e. (addressType = lowercase | checksum) - would also save us from inattention  mistakes & solidity type safety.

Just my 2 grains of salt, thanks a lot for producing viem.

**Comment by jxom** [comment]
Hey! Thanks for the feedback here. Those ideas are definitely very valuable!

> 1. decodeLogs crash sensitivity

Would you be able to provide a minimal reproduction here (usage + ABI)? Will be more than happy to look into this issue if you provide one!

> 2. Checksum is expensive

Definitely agree. We checksum by default as a "safety by default" thing, but for some usecases that rely heavily on perf, this may be useless (we can get another ~8x perf improvement by not checksumming!). Perhaps we can add a `checksumAddress: boolean` parameter to `decodeEventLogs`/`decodeAbiParameters`.

**Comment by NBMSacha** [comment]
Yes sharing here: 

```
export function parseSwapEvent(
  event: Log,
) {

switch (event.topics[0]) {
 case IZISWAP_DEPOSIT:
 case IZISWAP_WITHDRAW:
        const {
          // @ts-expect-error viem weird behavior
          args: {amountX, amountY},
        } = decodeEventLog({
          abi: event.topics[0] === IZISWAP_DEPOSIT ? IZISWAP_DEPOSIT_EVENT_ABI : IZISWAP_WITHDRAW_EVENT_ABI,
          data: event.data as any,
          topics: event.topics as any,
        });
  }
```

prior to this, I had: 
 abi:  IZISWAP_DEPOSIT_EVENT_ABI (no check on topic)
 
as both are identical, only the name is changing.

by the way the ts-expect-error is probably due to the fact that "Log" is from ethersjs, thus had to as any everything, thus viem is throwing an error.


For the checksum side of the thing, I'd say a general env / client level setting could be a huge feature (for us at least) - a parameter to decodeLog could be good tho.

**Comment by NBMSacha** [comment]
I love your website jxom.io btw. Amazing lighthouse score, congrats
